src/app/components/RulePicker.tsx

```diff
@@ -0,0 +1,172 @@
+  setMonth: (month: number) => void;
+}
+
+export default function RulePicker({ setDay, setMonth }: RulePickerProps) {
+  const [searchQuery, setSearchQuery] = useState("");
+  const [debouncedSearchQuery, setDebouncedSearchQuery] = useState("");
+
+  // Debounce search query
+  useEffect(() => {
+    const timer = setTimeout(() => {
+      setDebouncedSearchQuery(searchQuery);
+    }, 150);
+
+    return () => clearTimeout(timer);
+  }, [searchQuery]);
+
+  // Memoize the combined rules
+  const rules = useMemo(() => {
+    // Convert shared rules format to match regular rules
+    const formattedSharedRules = sharedRules.map((rule) => ({
+      rule: rule.rule,
+      description: rule.description,
+      notes: rule.notes,
+      date: rule.date,
+    }));
+
+    // Convert regular rules to include optional date property
+    const formattedRegRules = regRules.map((rule) => ({
+      rule: rule.rule,
+      description: rule.description,
+      notes: rule.notes,
+      date: undefined as string | undefined,
+    }));
+
+    // Combine both rule sets
+    return [...formattedRegRules, ...formattedSharedRules];
+  }, []);
+
+  // Memoize rule appearance status for all rules
+  const ruleAppearanceMap = useMemo(() => {
+    const map = new Map<
+      string,
+      { hasAppeared: boolean; firstOccurrence: Date | null }
+    >();
+
+    const startDate = new Date(2025, 0, 1); // Jan 1, 2025
+    const today = Time.debugMode ? new Date(2025, 11, 31) : new Date();
+
+    rules.forEach((rule) => {
+      let currentDate = new Date(startDate);
+      let hasAppeared = false;
+      let firstOccurrence: Date | null = null;
+
+      while (currentDate <= today && !hasAppeared) {
+        const daySchedule = Schedule.on(currentDate);
+        if (daySchedule.rule.rule === rule.rule) {
+          hasAppeared = true;
+          firstOccurrence = new Date(currentDate);
+        }
+        currentDate.setDate(currentDate.getDate() + 1);
+      }
+
+      map.set(rule.rule, { hasAppeared, firstOccurrence });
+    });
+
+    return map;
+  }, [rules]);
+
+  // Filter rules based on debounced search query
+  const filteredRules = useMemo(() => {
+    if (debouncedSearchQuery.trim() === "") {
+      return []; // Show nothing when search is empty
+    }
+
+    return rules.filter(
+      (rule) =>
+        rule.rule.toLowerCase().includes(debouncedSearchQuery.toLowerCase()) ||
+        rule.description
+          .toLowerCase()
+          .includes(debouncedSearchQuery.toLowerCase())
+    );
+  }, [debouncedSearchQuery, rules]);
+
+  const handleSearchChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    setSearchQuery(e.target.value);
+  };
+
+  const handleSelectRule = useCallback(
+    (rule: RuleItem) => {
+      const appearance = ruleAppearanceMap.get(rule.rule);
+
+      if (appearance?.hasAppeared && appearance.firstOccurrence) {
+        setMonth(appearance.firstOccurrence.getMonth());
+        setDay(appearance.firstOccurrence.getDate());
```

Comment:
After testing, I found the experience better if we clear the search query after selecting an option.

```suggestion
        setDay(appearance.firstOccurrence.getDate());
        setSearchQuery("");
```